ls1intum/Artemis
# [Enhancement] query speedups and exam tutor leaderboards

### Checklist

### Motivation and Context


Resolves https://github.com/ls1intum/Artemis/issues/1890
The loading time of the assessment-dashboard is still pretty long. (Now 6s instead of 30s, but there is still room for improvement.)
This is a followup to https://github.com/ls1intum/Artemis/pull/3001.
Additionally, as the tutor-leaderboard stats are no longer fetched from views, adding the leaderboards for the exam dashboards was possible.

### Description

- improved query performance in different places
- added tutor leaderboards in the exams assessment-dashboard and the exams exercise-assessment-dashboard:
![image](https://user-images.githubusercontent.com/33342534/110801840-e76aef80-827d-11eb-8570-48198a155def.png)
![image](https://user-images.githubusercontent.com/33342534/110801902-fbaeec80-827d-11eb-87a1-2c6e31bee9a9.png)



### Steps for Testing


1. Go to a course with exam. Go to the assessment dashboard and check if the numbers of the dashboard add up with the numbers in the participation graphs.<img src="https://user-images.githubusercontent.com/33342534/110802107-2a2cc780-827e-11eb-9a50-075f525764db.png" width="300">

2. No navigate to the exercise-assessment-dashboard and check the same.
